homeassistant/util/unit_system.py

```diff
@@ -121,6 +121,11 @@ def is_metric(self) -> bool:
         """Determine if this is the metric unit system."""
         return self.name == CONF_UNIT_SYSTEM_METRIC
 
+    @property
+    def is_us_customary(self) -> bool:
+        """Determine if this is the US Customary unit system."""
+        return self.name == CONF_UNIT_SYSTEM_IMPERIAL
+
```

Comment:
I cannot find a way to do a positive assert without adding this property to UnitSystem.
Although not optimal, this will help us to remove all references to `IMPERIAL_SYSTEM` in the components.

This is an alternative to https://github.com/home-assistant/core/pull/80180